packages/editor/src/components/rich-text/core-tokens/image/index.js

```diff
@@ -35,7 +35,11 @@ export const settings = {
 		);
 	},
 
-	save( { id, url, alt, width } ) {
+	save( { id, url, alt, width, filename } ) {
+		if ( ! alt ) {
+			alt = sprintf( __( 'This image has an empty alt attribute; its file name is "%s"' ), filename );
```

Comment:
The issue #1520 is only about adding this text to images in editing context. The `save()` method is what gets saved into the database and is visible on the front end. So this change seems wrong?

Reply:
I can change this to only be the edit context. I set the alt if it is empty on the save method because from my understanding, screen readers will try and read the full image url if there is no alt attribute defined.

Reply:
Related chat on Slack where this has been discussed: https://wordpress.slack.com/archives/C02SX62S6/p1540406745000100

tldr;: empty alt texts are fine, suggested change should only apply in editing context.